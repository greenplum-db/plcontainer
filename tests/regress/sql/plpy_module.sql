-- Check if the "plpy" module is created
CREATE OR REPLACE FUNCTION check_plpy_created() RETURNS bool AS $$
# container: plc_python_shared
import sys
return 'plpy' in sys.modules
$$ language plcontainer;

SELECT check_plpy_imported();

-- Check if 'plpy' is a built-in module
CREATE OR REPLACE FUNCTION check_plpy_builtin_module() RETURNS bool AS $$
# container: plc_python_shared
import sys
return 'plpy' in sys.builtin_module_names
$$ LANGUAGE plcontainer;

SELECT check_plpy_builtin_module();

CREATE OR REPLACE FUNCTION dir_plpy() RETURNS SETOF text AS $$
# container: plc_python_shared
return dir(plpy)
$$ LANGUAGE plcontainer;

SELECT att_name FROM dir_plpy() AS att_name WHERE att_name NOT LIKE '\_%' 
ORDER BY att_name;
