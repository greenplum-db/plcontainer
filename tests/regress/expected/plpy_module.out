-- Check if 'plpy' is an imported module
CREATE OR REPLACE FUNCTION check_plpy_imported() RETURNS bool AS $$
# container: plc_python_shared
import sys
return 'plpy' in sys.modules
$$ language plcontainer;
SELECT check_plpy_imported();
 check_plpy_imported 
---------------------
 t
(1 row)

-- Check if 'plpy' is a built-in module
CREATE OR REPLACE FUNCTION check_plpy_builtin_module() RETURNS bool AS $$
# container: plc_python_shared
import sys
return 'plpy' in sys.builtin_module_names
$$ LANGUAGE plcontainer;
SELECT check_plpy_builtin_module();
 check_plpy_builtin_module 
---------------------------
 t
(1 row)

CREATE OR REPLACE FUNCTION dir_plpy() RETURNS SETOF text AS $$
# container: plc_python_shared
return dir(plpy)
$$ LANGUAGE plcontainer;
SELECT att_name FROM dir_plpy() AS att_name WHERE att_name NOT LIKE '\_%' 
ORDER BY att_name;
    att_name    
----------------
 debug
 error
 Error
 execute
 fatal
 Fatal
 info
 log
 notice
 prepare
 quote_ident
 quote_literal
 quote_nullable
 SPIError
 spiexceptions
 subtransaction
 warning
(17 rows)

